packages/core/src/middleware/actions.ts

```diff
@@ -47,13 +68,33 @@ export const middlewareActions =
       customMiddleware: overrides.customMiddleware ?? noopMiddleware,
       dummyPaymasterAndData: overrides.paymasterAndData?.dummyPaymasterAndData
         ? async (struct, { account }) =>
-            account.getEntryPoint().version === "0.6.0"
+            account.getEntryPoint().version === "0.7.0"
               ? {
                   ...struct,
-                  paymasterAndData:
-                    overrides.paymasterAndData!.dummyPaymasterAndData(),
+                  ...((isHex(
+                    overrides.paymasterAndData!.dummyPaymasterAndData()
+                  )
+                    ? parsePaymasterAndData(
+                        overrides.paymasterAndData!.dummyPaymasterAndData() as Hex
```

Comment:
If you put `overrides.paymasterAndData!.dummyPaymasterAndData()` into a temp variable, you wouldn't need these casts because `isHex` would work as a type guard (plus you wouldn't need to keep writing this large expression over and over). I'd suggest:
```ts
overrides.paymasterAndData?.dummyPaymasterAndData
        ? async (struct, { account }) => {
            const data = overrides.paymasterAndData!.dummyPaymasterAndData();
            return account.getEntryPoint().version === "0.7.0"
              ? {
                  ...struct,
                  ...(isHex(data)
                    ? parsePaymasterAndData(data)
                    : concatPayamsterAndData(data)
                  ),
                }
                // ...
```
and maybe consider turning some of ternaries into if-else blocks for readability.

Reply:
> `isHex would work as a type guard `

Actually it doesn't which is why had to cast...

Reply:
It didn't work before because you're evaluating `dummyPaymasterAndData()` multiple times, so the type system can't tell that it's the same value. `isHex` should work as a type guard if you extract the value, its signature is
```ts
function isHex(
  value: unknown,
  { strict = true }: { strict?: boolean } = {},
): value is Hex;
```

Reply:
Here's what it looks like without the casts using `isHex` as a type guard, if you want it. Verified locally.
```ts
      dummyPaymasterAndData: overrides.paymasterAndData?.dummyPaymasterAndData
        ? async (struct, { account }) => {
            const data = overrides.paymasterAndData!.dummyPaymasterAndData();
            const paymasterOverrides =
              account.getEntryPoint().version === "0.7.0"
                ? isHex(data)
                  ? parsePaymasterAndData(data)
                  : data
                : {
                    paymasterAndData: isHex(data)
                      ? data
                      : concatPaymasterAndData(data),
                  };
            return { ...struct, ...paymasterOverrides };
          }
        : defaultPaymasterAndData,
```